From 73fdda3de788787b4e4941562a390f79e96c5a00 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Sun, 21 Jan 2024 22:18:50 -0800
Subject: [PATCH 1/2] Use std::string for processing environment variables

wxString cannot handle charset conversion correctly while std::string can.
The wxWidgets upstream encourages using std::string whenever possible as well.

Ref: https://docs.wxwidgets.org/3.0/classwx_string.html#details
---
 common/pgm_base.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/pgm_base.cpp b/common/pgm_base.cpp
index 91653ef4b1..c9cb71e25e 100644
--- a/common/pgm_base.cpp
+++ b/common/pgm_base.cpp
@@ -430,7 +430,7 @@ bool PGM_BASE::InitPgm( bool aHeadless, bool aSkipPyInit, bool aIsUnitTest )
         wxPGInitResourceModule();
 
 #ifndef __WINDOWS__
-    if( wxString( wxGetenv( "HOME" ) ).IsEmpty() )
+    if( std::string( wxGetenv( "HOME" ) ).empty() )
     {
         DisplayErrorMessage( nullptr, _( "Environment variable HOME is empty.  "
                                          "Unable to continue." ) );
-- 
2.39.1

